### PR TITLE
handle PBS resource directives by transalting them to SBATCH ones

### LINKED ISSUE #49: incorrect resource requests are silently ignored
Submitting this job script to `joltik`:

```bash
$ cat test_joltik.sh
#!/bin/bash
#PBS -l nodes=1,ppn=8,gpus=1

taskset -p $$
nvidia-smi
```

results in:

```
$ cat test_joltik.sh.o40004288
pid 181793's current affinity mask: 10000
No devices were found
```

So, only one single core was used, and no GPUs were available.

The job script is wrong (it should use `nodes=1:ppn=8:gpus=1` (`:` rather than `,`), but it is accepted anyway, and the incorrectly specified resource requests are silently ignored.

### LINKED ISSUE #51: "#PBS -l nodes=X:ppn=Y:gpus=1" in job script results in "Invalid node name specified"
(when submitting to `joltik`, of course)

The problem here seems to be that the `#PBS` header lines are processed by Slurm itself, which doesn't know about `gpus=1` as a node resource?

The error message produced here is very confusing, we should at least try to mitigate that on short notice (by checking whether `gpus=N` is used in a `#PBS` line in the job script.

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #51 
Fixes #49

### CODE REVIEW

**Comment by hajgato** [comment]
@stdweird version bump?

File: files/t/qsub-commands.t [hunk 1]
```diff
@@ -121,15 +124,24 @@ is(join(" ", @$newcommand), $txt, "expected command after parse_script without e
...
     ($newtxt, $newcommand) = parse_script($stdin, $command, $defaults);
     is(join(" ", @$newcommand),
        "$sbatch --nodes=2 --ntasks=8 --ntasks-per-node=4 --chdir=$ENV{HOME} --export=NONE --get-user-env=60L",
        "expected command after parse_script with eo");
-    is($newtxt, "#!/some/shell\n#\n#PBS -l abd -o ".getcwd."/stdout.%A..%A\n#\n#PBS -e /abc -N def\ncmd\n",
+    is($newtxt, "#!/some/shell\n#\n#SBATCH --nodes=123\n#PBS -o ".getcwd."/stdout.%A..%A\n#\n#PBS -e /abc -N def\ncmd\n",
        "PBS_JOBID replaced");
 }
 
+my $stdin = "#!/bin/something\n#PBS -l nodes=123:ppn=456:gpus=1 -o stdout -l mem=112233\n#\n#PBS -l gpus=2 -l mem=223344 -N def\ncmd\n";
```
Comment (boegel) [inline files/t/qsub-commands.t]: Maybe also test with something like `nodes=3:ppn=5:gpus=3` (should result in `--gres=gpu:3`, not `gpu:9`)?

Comment (boegel) [reply files/t/qsub-commands.t]: And `nodes=X:ppn=Y,gpus=Z` is also valid input, so test with that too?

Comment (stdweird) [reply files/t/qsub-commands.t]: no real need to test that, it's already tested as part of the commandline parsing

**Pull request merged**